src/components/store/AddCartButton/index.tsx

```diff
@@ -0,0 +1,105 @@
+import { Dropdown } from '@/components/common';
+import { useId } from 'react';
+import styles from './style.module.scss';
+
+interface AddCartButtonProps {
+  currSize: string | undefined;
+  inStock: boolean;
+  inCart: boolean;
+  lifetimeRemaining: number;
+  monthlyRemaining: number;
+  amountToBuy: number;
+  onCartChange: (inCart: boolean) => void;
+  onAmountChange: (amountToBuy: number) => void;
+}
+
+const AddCartButton = ({
+  currSize,
+  inStock,
+  inCart,
+  onCartChange,
+  lifetimeRemaining,
+  monthlyRemaining,
+  amountToBuy,
+  onAmountChange,
+}: AddCartButtonProps) => {
+  const myID = useId();
+
+  const maxCanBuy = Math.min(20, Math.min(lifetimeRemaining, monthlyRemaining));
+
+  const validText = inStock ? (
+    <p className={styles.valid}>This item is in stock.</p>
+  ) : (
+    <p className={styles.error}>This item is out of stock.</p>
+  );
+
+  let buyButtonText = 'Add to Cart';
+
+  if (inCart) {
+    buyButtonText = 'Remove from Cart';
+  } else if (maxCanBuy === 0) {
+    buyButtonText = 'Limit Reached';
+  } else if (currSize === undefined) {
+    buyButtonText = 'Select a Size';
+  } else if (inStock) {
+    buyButtonText = 'Add to Cart';
+  } else {
+    buyButtonText = 'Out of Stock';
+  }
+
+  const optionArr: Array<{ value: string; label: string } | '----'> = [];
+
+  if (maxCanBuy === 0) {
+    optionArr.push({ value: `0`, label: `0` });
+  }
+
+  for (let i = 1; i <= maxCanBuy; i += 1) {
+    optionArr.push({ value: `${i}`, label: `${i}` });
+  }
+
+  optionArr.push('----');
```

Comment:
What is the purpose of the final line?

![image](https://github.com/acmucsd/membership-portal-ui-v2/assets/22133785/b2c71c52-e9e1-4ce1-8044-0d82457c51e6)